File: CSharpCodeBase/aiactions/ai_selectweapon.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class ai_selectweapon :ainode{
 // Select enemy using ai vision
 public void init(self, entity);
   AINode.init(self, entity);
 }})
 public void Visit(){
   if(this.status == NODE_READY  ){ 
     var nextWeapon = this.entity.weaponSequencer:GetNextWeapon();
     this.entity.weaponContainer:SetWeapon(nextWeapon);
     this.status = NODE_SUCCESS;
   }
   return this.status;
 }
 }}

File: CSharpCodeBase/aiactions/ai_selectweapon.lua
local Class = require 'utils.hump.class'
local AINode = require 'utils.ailib.ainode'

-- Select enemy using ai vision
local AI_SelectWeapon = Class({__includes=AINode, init=function(self, entity)
  AINode.init(self, entity)
end})

function AI_SelectWeapon:Visit()
  if self.status == NODE_READY then
    local nextWeapon = self.entity.weaponSequencer:GetNextWeapon()
    self.entity.weaponContainer:SetWeapon(nextWeapon)
    self.status = NODE_SUCCESS
  end
  return self.status
end

return AI_SelectWeapon
